src/Presenter/Store/Modules/ConfigurationModule.php

```diff
@@ -147,4 +155,33 @@ private function isFundingSourceCardEnabled()
 
         return false;
     }
+
+    /**
+     * Checks if any currencies are enabled for which PayPal doesn't support decimal values
+     * Returns error message with listed currencies that have to be configured correctly
+     *
+     * @return array
+     */
+    private function checkNonDecimalCurrencies()
+    {
+        $nonDecimalCurrencies = ['HUF', 'JPY', 'TWD'];
+
+        $currencies = \Currency::getCurrencies(false, true);
+
+        $misConfiguredCurrencies = [];
+
+        foreach ($currencies as $currency) {
+            if (in_array($currency['iso_code'], $nonDecimalCurrencies) && (int) $currency['precision'] !== 0) {
+                $misConfiguredCurrencies[] = $currency['iso_code'];
+            }
+        }
+
+        return [
+            'showError' => !empty($misConfiguredCurrencies),
+            'errorMessage' => sprintf(
+                $this->module->l('Attention: you have activated %s currencies, you need to configure those currencies to use 0 decimals as Paypal does not support decimals for those currencies'),
```

Comment:
```suggestion
                $this->module->l('Attention: you have activated %s currencies, you need to configure those currencies to use 0 decimals as Paypal does not support decimals for those currencies', 'configurationmodule'),
```
Legacy translations system needs to have filename in lowercase in this case or translations will never be found